executions/graphql-kotlin-dataloader-instrumentation/build.gradle.kts

```diff
@@ -9,6 +9,7 @@ dependencies {
     api(libs.graphql.java) {
         exclude(group = "com.graphql-java", module = "java-dataloader")
     }
+    testImplementation(projects.graphqlKotlinServer)
```

Comment:
why do you need to bring this dependency?

Reply:
It's to have access to the [dataFetchingEnvironmentExtensions](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/extensions/dataFetchingEnvironmentExtensions.kt) in the tests.

Reply:
I think it is somewhat problematic as you end up with cyclic dependency -> i.e. `graphql-kotlin-server` depends on `graphql-kotlin-dataloader-instrumentation` as API dependency ([link](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/servers/graphql-kotlin-server/build.gradle.kts#L13)) and we depend here on the server as a test dependency.

Guessing Gradle is smart enough to handle the API vs test dependency.... but I'd avoid it and either copy over the extension locally to one of the tests or simply do what extension does....